### ISSUE TITLE
'BUG': get_loader() returns None  when loading bst project

### ISSUE THREAD

**Issue by harrysarson** [apache/buildstream#1976]
Hi we have seen a bug whilst loading a buildstream project that uses junctions. The bug ends up with `AttributeError: 'NoneType' object has no attribute '_load_file'` in `buildstream/_loader/loader.py` presumably because `get_loader()` returns `None` here https://github.com/apache/buildstream/blob/6d16e3a746d0f8fc4b1873f0bf82e4ddd90d7f85/src/buildstream/_loader/loader.py#L189.

I have seen https://github.com/apache/buildstream/pull/1975 which seems to make `get_loader()` returning `None` more robust. However, we are struggling to reproduce so are not able to confirm whether that issue fixes the bug.

**Comment by harrysarson** [comment]
I have managed to reproduce this bug (unfortunately I cannot share the code). I have confirmed that applying the fix in #1975 fixes the issue. 

To reproduce I needed:

1. Four bst projects `A`, `B`, `C`, `D`. `A` depends on `B` via a junction, `B` depends on `C` via a junction and `C` depends on `D` via a junction.
2. `A` uses link elements to refer to `C` and `D`.
3. `A` overrides an element in `D`

I haven't managed to reproduce on anything simpler than four projects (so three junctions).
